cannon/mipsevm/tests/helpers.go

```diff
@@ -50,31 +50,50 @@ func multiThreadElfVmFactory(t require.TestingT, elfFile string, po mipsevm.Prei
 	return fpvm
 }
 
+type ThreadProofEncoder func(po mipsevm.PreimageOracle, stdOut, stdErr io.Writer, log log.Logger, opts ...testutil.StateOption) []byte
+
+func singalThreadProofEncoder(po mipsevm.PreimageOracle, stdOut, stdErr io.Writer, log log.Logger, opts ...testutil.StateOption) []byte {
+	return nil
+}
+
+func multiThreadProofEncoder(po mipsevm.PreimageOracle, stdOut, stdErr io.Writer, log log.Logger, opts ...testutil.StateOption) []byte {
```

Comment:
The only parameter we need here is `opts` to reconstruct the state. Let's remove the others.